### PR TITLE
Add the IS_PRO switch to OS PR Jenkinsfile for snyk code check run

### PR DESCRIPTION
**Pull request by GCRev**
See: https://github.com/rstudio/rstudio-pro/issues/5606

### Intent

Pare down the output of the Snyk code scan for the OS repo.

### Approach

Add in an extra check before posting the output and use a link to the Jenkins job instead.

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

File: jenkins/Jenkinsfile.pull-request [hunk 1]
```diff
@@ -54,8 +54,21 @@ pipeline {
...
+        script {
+          exitCode = sh(returnStatus: true, script: "snyk code test ${TMP_SNYK_PATH} | sed '/^Testing/d' | sed '/./,\$!d' > ${TMP_SNYK_PATH}/output.txt")
+          // - exit code 0: sucessful scan, no vulns identified in code
+          // - exit code 1: unsuccessful scan, code had vulns (real or false positive)
+          // - exit code 3: scan was not-applicable to files. This is typical for docs or NEWS commits
+          // 
+          // In the case of code 3, we don't want to post anything in order to avoid generating confusion
+          if (exitCode != 3) {
+            status = exitCode == 0 ? "success" : "failure"
+            text = !IS_PRO ? JOB_URL : sh(returnStdout: true, script: "jq -Rs . ${TMP_SNYK_PATH}/output.txt")
```
Comment (sharon-wang) [inline jenkins/Jenkinsfile.pull-request:68]: Is it possible to drill in further than the PR jenkins `JOB_URL` (do we have access to / can we construct the rest of the url to the snyk results)?

It takes a few clicks to get to the snyk results. If not, maybe we can add some dev docs on how to navigate to the snyk results from the PR job url page.

Comment (GCRev) [reply jenkins/Jenkinsfile.pull-request:68]: I'll investigate whether we can make the right assumptions about the url to get a link directly to that part of the output. Also I probably need to `cat` the output to the console as well.

**Comment by GCRev** [comment]
@sharon-wang I've gotten the link to be closer. I could hard-code something with the url to get into the exact node of the pipeline console, but I'm not sure that's super stable.

**Review by sharon-wang — APPROVED** [review]
Nice!! I think the scan results were something like 3 clicks away, now only 1 so that's great 👌

**Pull request merged**